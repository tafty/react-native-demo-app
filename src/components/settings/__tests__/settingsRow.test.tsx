import 'react-native'

import { fireEvent, render } from 'react-native-testing-library'

import React from 'react'
import SettingsRow from '../settingsRow'

describe('SettingsRow', () => {
  it('should render the row', () => {
    const title = 'The title'
    const onPress = jest.fn()

    const { getByTestId, toJSON } = render(
      <SettingsRow title={title} onPress={onPress} />,
    )

    expect(getByTestId('The title-button-label').props.children).toBe(
      title.toUpperCase(),
    )

    fireEvent.press(getByTestId('The title-button'))
    expect(onPress).toBeCalled()

    expect(toJSON()).toMatchSnapshot()
  })
})
