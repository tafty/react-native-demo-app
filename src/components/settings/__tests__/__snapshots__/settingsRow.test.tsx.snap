// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsRow should render the row 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginVertical": 8,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 124, 255, 1)",
        "borderColor": "rgba(0, 124, 255, 1)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 16,
      }
    }
    testID="The title-button"
  >
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "flex": 0,
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="The title-button-label"
    >
      THE TITLE
    </Text>
  </View>
</View>
`;
